### DISCUSSION TITLE
[memory about consumer]Is it normal for memory to grow slowly?

### DISCUSSION THREAD

**Discussion by Mcefor19** [edenhill/librdkafka#3681] [Q&A] [answered]
First,i think the librdkafka is a good lib,i like it.
Recently I encountered some confusion in the process of using.

-  kafka version:kafka_2.11-0.11.0.2
- kafka client :c++

I started a consumer and found that running it for a longer period of time slowly grows its memory(run for 3 days,1984K->2084K,produce a piece of data every 2ms).
The code:
![图片](https://user-images.githubusercontent.com/23627614/150252651-80a4442b-51b6-4ea7-a7a7-c70c9b0420f4.png)

I read the  https://github.com/edenhill/librdkafka/wiki/FAQ#explain-the-consumers-memory-usage-to-me.
And my question is is this phenomenon of memory growth normal?
Have you encountered such problems in the process of using it? Can you give me some help? Thank you so much!

**Comment by edenhill** [comment, marked as answer]
What version of librdkafka?

There are no known memory leaks in up to date librdkafka releases.
Suggest using a memory profiler (asan, valgrind, ..) to check for memory leaks.
If using valgrind you may also want to use the suppressions file in `tests/librdkafka.suppressions` to silence some false positives.

**Reply by Mcefor19** [reply]
Thank you very much for your reply.
I downloaded the code from https://github.com/edenhill/librdkafka,and compiled it into a library
